test.js

```diff
@@ -14,9 +14,88 @@ test('all errors are instances of LevelUPError', function (t) {
   t.end()
 })
 
+test('error with message has expected properties', function (t) {
+  const error = new errors.ReadError('foo')
+
+  t.is(error.type, 'ReadError')
+  t.is(error.name, 'ReadError')
+  t.is(error.cause, undefined)
+  t.is(error.message, 'foo')
+  t.end()
+})
+
+test('error without message has expected properties', function (t) {
+  const error = new errors.ReadError()
+
+  t.is(error.type, 'ReadError')
+  t.is(error.name, 'ReadError')
+  t.is(error.cause, undefined)
+  t.is(error.message, '')
+  t.end()
+})
+
+test('error with cause has expected properties', function (t) {
+  const cause = new Error('foo')
+  const error = new errors.ReadError(cause)
+
+  t.is(error.type, 'ReadError')
+  t.is(error.name, 'ReadError')
+  t.ok(error.cause === cause)
+  t.is(error.message, 'foo')
+  t.end()
+})
+
+test('error with message and cause has expected properties', function (t) {
+  const cause = new Error('foo')
+  const error = new errors.ReadError('bar', cause)
+
+  t.is(error.type, 'ReadError')
+  t.is(error.name, 'ReadError')
+  t.ok(error.cause === cause)
+  t.is(error.message, 'bar')
+  t.end()
+})
+
 test('NotFoundError has special properties', function (t) {
   const error = new errors.NotFoundError()
-  t.equal(error.notFound, true)
-  t.equal(error.status, 404)
+  t.is(error.notFound, true)
+  t.is(error.status, 404)
+  t.end()
+})
+
+test('error message is writable for flexibility', function (t) {
+  const error = new errors.WriteError('foo')
+  error.message = 'Got error: ' + error.message
```

Comment:
is this potentially dangerous (as you expect errors not to behave like this), and not worth the convenience? 🤔

Reply:
It aligns with the behavior of:

```
$ node
> Object.getOwnPropertyDescriptor(new Error('foo'), 'message')
{ value: 'foo', writable: true, enumerable: false, configurable: true }
````

Reply:
ah! If we can mirror node core conventions then +1